sdk/core/core-util/package.json

```diff
@@ -65,15 +65,16 @@
     "tslib": "^2.2.0"
   },
   "devDependencies": {
+    "@azure-tools/vite-plugin-browser-test-map": "^1.0.0",
     "@azure/dev-tool": "^1.0.0",
     "@azure/eslint-plugin-azure-sdk": "^3.0.0",
-    "@azure-tools/vite-plugin-browser-test-map": "^1.0.0",
     "@microsoft/api-extractor": "^7.31.1",
     "@types/node": "^18.0.0",
     "@vitest/browser": "^1.2.1",
     "eslint": "^8.0.0",
     "playwright": "^1.39.0",
     "rimraf": "^3.0.0",
+    "type-plus": "^7.6.0",
```

Comment:
What is this for?

Reply:
I'm taking only type imports from this library, since it has a bunch of utilities. I ended up having to take a dev dependency on this elsewhere in the repo, as it's [shockingly difficult](https://github.com/unional/type-plus/blob/c0c5694e5f43c926be37d972c5e9ac7953ecfd53/packages/type-plus/src/equal/equal.ts#L38-L71) to check if, for example, two types are identical. It gets even weirder as you dig deeper. In fixing the bug you mentioned, I ended up having to use this in the definition of the type itself, and it made the LSP hover type more verbose in the tests. Typically, I have to play around with this sort of thing to stop `tsserver` from expanding the type definition without affecting the semantics.

In any case, this is something we can vendor if need be, especially since the bloat is all compiled away by `tsc`. But it's a point against this solution.